cpp-client/src/polygon.cpp

```diff
@@ -82,28 +82,35 @@ bool Polygons::findFirstNonMatching(const Polygons &polygons, int &polyIndex, in
   return false;
 }
 
-Polygons Polygons::fromJson(const std::string &json) {
+Polygons Polygons::fromJson(const std::string &json, const std::string &key) {
   try {
     nlohmann::json j = nlohmann::json::parse(json);
+    if (!key.empty() && j.find(key) != j.end()) {
+      j = j[key];
+    }
 
     Polygons polygons;
     for (auto poly : j) {
       Polygons::Polygon polygon;
-      for (auto pt : poly) {
-        Polygons::Point point;
-        for (auto n : pt) {
-          point.push_back(n);
+      if (!poly.empty()) {
```

Comment:
just curious,
does this check necessary?
If so what about line 93, do we need to check if "j" is empty?

Reply:
when polygon inside polygons have []... you still have to return like [] instead of [[]]